src/dataflow/src/server.rs

```diff
@@ -900,7 +900,15 @@ where
                     } else {
                         true // retain
                     }
-                })
+                });
+
+                // Respond with cancellation.
```

Comment:
I *think* this logic is incorrect, or out of sync with the coordinator at least. This will report a cancellation, even if there is no pending peek (e.g. perhaps because it has already been responded to and enqueued). This means that this worker will send both a peek response and a cancellation back, and the coordinator will count down twice.

I think the right thing to do is to send the cancellation only in the case that the peek was actually removed by this call (e.g. in the `// don't retain` case).